crates/bevy_ecs/src/query/state.rs

```diff
@@ -309,6 +350,7 @@ impl<D: QueryData, F: QueryFilter> QueryState<D, F> {
             component_access,
             matched_tables: Default::default(),
             matched_archetypes: Default::default(),
+            sync_by_observer: false,
```

Comment:
Is this a temporary measure?

Reply:
Right, that might be something to change for a follow-up PR then. I don't like having two ways to sync a QueryState.